### PR TITLE
Fix sql datetime formating 185

### PR DESCRIPTION
**Pull request by gbrian**
@xrmx Please review ;)

About test should be a better way, right?

**Note about ORACLE**: who did that one? Not sure if it was better the old version, new version has been tested with http://sqlfiddle.com/#!15/27c44/2
`SELECT TO_TIMESTAMP('2016-07-08 09:10:11.131456', 'YYYY-MM-DD HH24:MI:SS.ff6');`

https://github.com/airbnb/caravel/issues/185

### CODE REVIEW

File: caravel/models.py [hunk 1]
```diff
@@ -959,7 +959,7 @@ def sqla_col(self):
             col = literal_column(self.expression).label(name)
         return col
 
-    def dttm_sql_literal(self, dttm):
+    def dttm_sql_literal(self, dttm, field_type):
```
Comment (xrmx) [inline caravel/models.py]: how is this supposed to work if you are not updating the callers?

**Comment by xrmx** [comment]
@gbrian any chance you can rebase on a branch following caravel master instead of whatever you have in yours? You have a dozen unrelated commits for not good reason.

BTW you should add "Fix #185" at the end of the commit message so bug is closed automatically.

File: caravel/models.py [hunk 2]
```diff
@@ -970,27 +970,21 @@ def dttm_sql_literal(self, dttm):
...
-                'presto': default,
-                'sqlite': default,
-            }
-            for k, v in d.items():
-                if self.table.database.sqlalchemy_uri.startswith(k):
-                    return v
-            return default
+            uri = self.table.database.sqlachemy_uri
+            if uri.startswith('oracle'):
+               return "TO_TIMESTAMP('{}', 'YYYY-MM-DD HH24:MI:SS.ff6')"\
```
Comment (xrmx) [inline caravel/models.py]:
```
return (
    "TO_TIMESTAMP"...
..
)
```
so we can avoid the ugly \

**Comment by xrmx** [comment]
If you move the tests in core_tests can we save some boilerplate? We have already a populated db there.

**Comment by mistercrunch** [comment]
Please setup travis to run against your fork and get the build working before sending in the PR. Also you may want to change your git workflow to leave less / comprehensive commits.

**Pull request closed without merging**